sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -28,6 +28,22 @@ import org.apache.spark.unsafe.types.{CalendarInterval, UTF8String}
 
 object IntervalUtils {
 
+  object IntervalUnit extends Enumeration {
+    type IntervalUnit = Value
+
+    val Nanosecond = Value(0, "nanosecond")
```

Comment:
What is the first value (here 0) used for?

Reply:
This is just be sure that enum values are ordered, so `Microsecond` < ... < `Second` < ... < `Year`.